### PR TITLE
Make rx/tx buffer capacity configurable via FIXConfig

### PR DESCRIPTION
**Pull request by jheusser**
Addressing #5 with defaults as before.

### CODE REVIEW

**Comment by jheusser** [comment]
Best to view with ignoring whitespace https://github.com/jvirtanen/philadelphia/pull/6/files?w=1

File: philadelphia/src/main/java/org/jvirtanen/philadelphia/FIXConfig.java [hunk 1]
```diff
@@ -7,260 +7,311 @@
...
+	 * @param version the protocol version
+	 * @param senderCompId the SenderCompID(49)
+	 * @param targetCompId the TargetCompID(56)
+	 * @param heartBtInt the HeartBtInt(108)
+	 * @param incomingMsgSeqNum the incoming MsgSeqNum(34)
+	 * @param outgoingMsgSeqNum the outgoing MsgSeqNum(34)
+	 * @param maxFieldCount the maximum number of fields in a message
+	 * @param fieldCapacity the field capacity
+	 * @param rxBufferCapacity the field receive buffer capacity
+	 * @param txBufferCapacity the field transmit buffer capacity
```
Comment (jvirtanen) [inline philadelphia/src/main/java/org/jvirtanen/philadelphia/FIXConfig.java]: These should probably say "the receive buffer capacity" and "the transmit buffer capacity".

File: philadelphia/src/main/java/org/jvirtanen/philadelphia/FIXConfig.java [hunk 2]
```diff
@@ -7,260 +7,311 @@
...
+		 * @return this instance
+		 */
+		public Builder setFieldCapacity(int fieldCapacity) {
+			this.fieldCapacity = fieldCapacity;
+
+			return this;
+		}
+
+		/**
+		 * Set the receive buffer capacity
```
Comment (jvirtanen) [inline philadelphia/src/main/java/org/jvirtanen/philadelphia/FIXConfig.java]: Can you add the full stop at the end of the sentence.

File: philadelphia/src/main/java/org/jvirtanen/philadelphia/FIXConfig.java [hunk 3]
```diff
@@ -7,260 +7,311 @@
...
+		 *
+		 * @param rxBufferCapacity the receive buffer capacity
+		 * @return this instance
+		 */
+		public void setRxBufferCapacity(int rxBufferCapacity) {
+			this.rxBufferCapacity = rxBufferCapacity;
+		}
+
+		/**
+		 * Set the transmit buffer capacity
```
Comment (jvirtanen) [inline philadelphia/src/main/java/org/jvirtanen/philadelphia/FIXConfig.java]: See above.

**Comment by jvirtanen** [comment]
Thanks, looks good. However, I would like to keep the four space indentation in `FIXConfig.java`. To fix the indentation, you can run, for example:

```
expand -t 4 philadelphia/src/main/java/org/jvirtanen/philadelphia/FIXConfig.java
```

**Comment by jheusser** [comment]
Have fixed the above and reformatted back to four space indentation.

File: philadelphia/src/main/java/org/jvirtanen/philadelphia/FIXConfig.java [hunk 4]
```diff
@@ -251,6 +282,26 @@ public Builder setFieldCapacity(int fieldCapacity) {
         }
 
         /**
+         * Set the receive buffer capacity.
+         *
+         * @param rxBufferCapacity the receive buffer capacity
+         * @return this instance
+         */
+        public void setRxBufferCapacity(int rxBufferCapacity) {
```
Comment (jvirtanen) [inline philadelphia/src/main/java/org/jvirtanen/philadelphia/FIXConfig.java]: These methods should return a `Builder`.

**Comment by jvirtanen** [comment]
Looking good, however, I noticed something I missed on the first round: the `Builder` methods should return `this`.

**Comment by jheusser** [comment]
This took a lot of attempt sorry, shouldn't have started this late Friday night :)

**Comment by jvirtanen** [comment]
Well, I'm always happy to get contributions. Thank you! :cake:

**Pull request merged**